pkgs/applications/misc/bazecor/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, appimageTools
+, fetchurl
+}:
+
+appimageTools.wrapAppImage rec {
+  pname = "bazecor";
+  version = "1.3.6";
+
+  src = appimageTools.extract {
+    inherit pname version;
+    src = fetchurl {
+      url = "https://github.com/Dygmalab/Bazecor/releases/download/v.${version}/Bazecor-${version}-x64.AppImage";
+      hash = "sha256-Mz7T/AAlyfMzdfy/ZV4AEP3ClTolwr2pPzkSCPL66/w=";
+    };
+
+    # Workaround for https://github.com/Dygmalab/Bazecor/issues/370
+    postExtract = ''
+      substituteInPlace \
+        $out/usr/lib/bazecor/resources/app/.webpack/main/index.js \
+        --replace \
+          'checkUdev=()=>{try{if(c.default.existsSync(f))return c.default.readFileSync(f,"utf-8").trim()===l.trim()}catch(e){console.error(e)}return!1}' \
+          'checkUdev=()=>{return 1}'
+    '';
+  };
+
+  # also make sure to update the udev rules in ./10-dygma.rules; most recently
+  # taken from
+  # https://github.com/Dygmalab/Bazecor/blob/v1.3.6/src/main/utils/udev.ts#L6
+
+  extraPkgs = p: (appimageTools.defaultFhsEnvArgs.multiPkgs p) ++ [
+    p.glib
+  ];
+
+  # Also expose the udev rules here, so it can be used as:
+  #   services.udev.packages = [ pkgs.bazecor ];
+  # to allow non-root modifications to the keyboards.
+
+  extraInstallCommands = ''
+    mv $out/bin/bazecor-* $out/bin/bazecor
```

Comment:
Forgot to ask, does this work for you? It stopped working for me after the 1.3.6 update and I had to change it to this: https://github.com/gvolpe/nix-config/commit/c60e2b1941ef36e7054d109b780c3823d14eed81

This is something that users won't be able to override without having to override the whole thing.

Reply:
Yes, it works well, what does not work for you? Does it fail and build or at run time?

Reply:
It fails at runtime, but now I double-checked and *it seems to be a flake-only issue*. Here's how I run it:

```
$ nix run .#bazecor
error: unable to execute '/nix/store/z4n94dzsb5l9w2skg1b1x47d87lif1zn-bazecor-1.3.6/bin/bazecor-1.3.6': No such file or directory
```

Here's where I have it declared on [my flake](https://github.com/gvolpe/nix-config/blob/master/flake.nix#L96): 

```nix
{
  packages.${system} = {
    inherit (pkgs) bazecor metals metals-updater;
  };
}
```

Adding an `apps` entry as follows fixes the issue. I guess it's because the name of the derivation includes the version.

```nix
{
  apps.${system}.bazecor = {
    type = "app";
    program = "${pkgs.bazecor}/bin/bazecor";
  };
}
```

Apologies for boycotting your PR, this is the last one I promise 😄

Reply:
Thanks, this is actually a symptom of me forgetting to set `meta.mainProgram`! Now done in the latest force push, such that your first approach should work out of the box (no `apps` necessary).